airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -237,42 +241,55 @@ def process(self) -> List["StreamProcessor"]:
         from_table = self.add_to_outputs(
             self.generate_json_parsing_model(from_table, column_names),
             self.get_model_materialization_mode(is_intermediate=True),
+            is_intermediate=True,
             suffix="ab1",
         )
         from_table = self.add_to_outputs(
             self.generate_column_typing_model(from_table, column_names),
             self.get_model_materialization_mode(is_intermediate=True, column_count=column_count),
+            is_intermediate=True,
             suffix="ab2",
         )
         if self.destination_sync_mode != DestinationSyncMode.append_dedup:
             from_table = self.add_to_outputs(
                 self.generate_id_hashing_model(from_table, column_names),
                 self.get_model_materialization_mode(is_intermediate=True, column_count=column_count),
+                is_intermediate=True,
                 suffix="ab3",
             )
             from_table = self.add_to_outputs(
                 self.generate_final_model(from_table, column_names),
                 self.get_model_materialization_mode(is_intermediate=False, column_count=column_count),
+                is_intermediate=False,
             )
         else:
+            if DestinationType.POSTGRES.value == self.destination_type:
+                # because of https://github.com/dbt-labs/docs.getdbt.com/issues/335, we have to use tables for postgres
+                forced_materialization_type = TableMaterializationType.TABLE
+            else:
+                forced_materialization_type = TableMaterializationType.VIEW
```

Comment:
VIEW materialization on Postgres and INCREMENTAL don't do well together

Reply:
I created this issue to gain a measurable impact of such changes in the future: https://github.com/airbytehq/airbyte/issues/7741